vortex-array/src/metadata.rs

```diff
@@ -122,26 +152,25 @@ where
     }
 }
 
-impl<'m, M> DeserializeMetadata<'m> for SerdeMetadata<M>
+impl<M> DeserializeMetadata for SerdeMetadata<M>
 where
-    M: serde::Deserialize<'m>,
+    M: Debug,
+    M: for<'m> serde::Deserialize<'m>,
 {
-    fn deserialize(metadata: Option<&'m [u8]>) -> VortexResult<Self> {
+    type Output = M;
+
+    fn deserialize(metadata: Option<&[u8]>) -> VortexResult<Self::Output> {
         let bytes =
             metadata.ok_or_else(|| vortex_err!("Serde metadata requires metadata bytes"))?;
-        Ok(SerdeMetadata(M::deserialize(
-            flexbuffers::Reader::get_root(bytes)?,
-        )?))
+        Ok(M::deserialize(flexbuffers::Reader::get_root(bytes)?)?)
     }
-}
 
-#[allow(clippy::use_debug)]
-impl<M> Display for SerdeMetadata<M>
-where
-    M: Debug,
-{
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{:?}", self.0)
+    #[allow(clippy::use_debug)]
+    fn format(metadata: Option<&[u8]>, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match Self::deserialize(metadata) {
+            Ok(m) => write!(f, "{:?}", m),
```

Comment:
would be nice to have actual display instead of just going through debug but it's a start

Reply:
It's what we did before for all the metadatas 🤷